src/test/java/com/meilisearch/integration/TasksTest.java

```diff
@@ -58,13 +58,13 @@ public void testClientGetTasks() throws Exception {
         for (Task task : tasks) {
             client.waitForTask(task.getUid());
 
-            assertNotNull(task.getStatus());
             assertNotNull(task.getStatus());
             assertTrue(task.getUid() >= 0);
             assertNotNull(task.getDetails());
             if (task.getType().equals("indexDeletion")) {
                 assertNotNull(task.getDetails().getDeletedDocuments());
             }
+            break;
         }
```

Comment:
I think you can just something like:
```
        Task task = tasks[0];
        client.waitForTask(task.getUid());

        assertNotNull(task.getStatus());
        assertTrue(task.getUid() >= 0);
        assertNotNull(task.getDetails());
        if (task.getType().equals("indexDeletion")) {
            assertNotNull(task.getDetails().getDeletedDocuments());
        }
```
It should work

Reply:
HI @alallema, thanks for the suggestion. That was my first approach, however with that I get a ArrayIndexOutOfBoundsException which is why I had to use the Break within a For Each loop

Reply:
This should not be the case except if this test is run before any task is processed. To make sure this does not happen, the solution is to add a task by running a `createIndex` or `AddDocument` request just before a `waitForTask()`:
```java
        String indexUid = "GetClientTasks";
        TaskInfo response = client.createIndex(indexUid);
        client.waitForTask(response.getTaskUid());
```